app/internal/wm/GioView.java

```diff
@@ -132,6 +134,64 @@ private void setCursor(int id) {
 		setPointerIcon(pointerIcon);
 	}
 
+    private void setBarColor(String t, int color, int luminance) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            return;
+        }
+
+        Window window = ((Activity) this.getContext()).getWindow();
+        if (window == null) {
+            return;
+        }
+
+        int insetsMask = 0;
+        int viewMask = 0;
+
+        if (t == "Status") {
+            insetsMask = WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS;
+            viewMask = View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
+
+            window.setStatusBarColor(color);
+        } else {
+            insetsMask = WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS;
+            viewMask = View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR;
+
+            window.setNavigationBarColor(color);
+        }
+
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            return;
+        }
+
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.Q) {
+            WindowInsetsController insetsController = window.getInsetsController();
+            if (insetsController == null) {
```

Comment:
Is insetsController ever null? If not, I'd rather leave a potential NPE than silently ignore a color change.

Reply:
The documentation say "This value may be null.", https://developer.android.com/reference/android/view/Window#getInsetsController(). 🤔